File: src/path/sliding/Move.java
package path.sliding;

public enum Move {
	SLIDE_UP(0, 1),
	SLIDE_DOWN(0, -1),
	SLIDE_LEFT(1, 0),
	SLIDE_RIGHT(-1, 0);
	
	public final int relPosX;
	public final int relPosY;
	
	Move(int x, int y) {
		relPosX = x;
		relPosY = y;
	}
}

File: src/main/scala/nl/calmamity/search/path/sliding/Move.scala
package nl.calmamity.search.path.sliding

/**
  * Defines the possible moves in a sliding puzzle. In any state, at most one tile can slide in a given direction, which
  * means that only the direction is required to define the move.
  */
object Move extends Enumeration {
	val SlideUp, SlideDown, SlideLeft, SlideRight = Value
	
	def determineRelativePosition(move: Value): (Int, Int) = {
		move match {
			case SlideUp =>
				(0, 1)
			case SlideDown =>
				(0, -1)
			case SlideLeft =>
				(1, 0)
			case SlideRight =>
				(-1, 0)
		}
	}
}
